src/Components/DataHandler/OrderActionLog/OrderActionLogDataHandler.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PayonePayment\Components\DataHandler\OrderActionLog;
+
+use Shopware\Core\Checkout\Order\OrderEntity;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+
+class OrderActionLogDataHandler implements OrderActionLogDataHandlerInterface
+{
+    public function __construct(
+        protected readonly EntityRepository $orderActionLogRepository
+    ) {
+    }
+
+    /**
+     * @param array<string, mixed> $request
+     * @param array<string, mixed> $response
+     */
+    public function createOrderActionLog(
+        OrderEntity $order,
+        array $request,
+        array $response,
+        Context $context
+    ): void {
+        $orderActionLog = [
+            'orderId' => $order->getId(),
+            'transactionId' => $response['txid'],
+            'referenceNumber' => $order->getOrderNumber(),
+            'request' => $request['request'],
+            'response' => $response['status'],
+            'amount' => $request['amount'],
+            'mode' => $request['mode'],
+            'merchantId' => $request['mid'],
+            'portalId' => $request['portalid'],
+            'requestDetails' => $request,
+            'responseDetails' => $response,
+            'requestDateTime' => new \DateTime(),
+        ];
+
+        $this->orderActionLogRepository->create([$orderActionLog], $context);
```

Comment:
please add a try-catch-block for the case, if any error occurs during persisting the data. 
log the error in a logfile.

So we make sure, that exceptions during logging will not interrupt any processes.

Reply:
Done